src/ClassicUO.Client/Game/Weather.cs

```diff
@@ -409,78 +894,675 @@ public void Draw(UltimaBatcher2D batcher, int x, int y, float layerDepth)
+                                            if (RandomHelper.RandomBool())
+                                            {
+                                                effect.NeverFade = true;
+                                                effect.FadeThreshold = 1.0f;
+                                            }
+                                            else
+                                            {
+                                                effect.NeverFade = false;
+                                                effect.FadeThreshold = 0.71f + RandomHelper.GetValue(0, 29) * 0.01f;
+                                            }
+                                            break;
+                                    }
+
+                                    continue; // Skip rendering, particle respawning
+                                }
+                            }
+
+                            // Calculate speed-based trail length
+                            float speedMagnitude = (float)Math.Sqrt(effect.SpeedX * effect.SpeedX + effect.SpeedY * effect.SpeedY);
+
+                            switch (rainStyle)
+                            {
+                                case RainRenderStyle.SmallDots:
+                                    // Multi-layer rendering for soft, glowing rain drops
+                                    int smallDotSize = (int)(3 * rainDepthProps.SizeMultiplier);
+                                    smallDotSize = Math.Max(2, smallDotSize);
+
+                                    // Base color for small rain drops - light blue
+                                    Color smallDotBaseColor = Color.Lerp(Color.LightBlue, rainDepthProps.ColorTint, 0.1f);
+                                    // Boost alpha for visibility
+                                    float smallDotBoostedAlpha = Math.Max(0.8f, rainDepthProps.AlphaMultiplier);
+
+                                    // Outer layer (largest, most transparent - edge)
+                                    if (smallDotSize >= 2)
+                                    {
+                                        int outerSize = smallDotSize;
+                                        float outerAlpha = smallDotBoostedAlpha * 0.25f; // 25% of base alpha for outer edge
+                                        Color outerColor = smallDotBaseColor * outerAlpha;
+
+                                        Rectangle outerRect = new Rectangle(
+                                            newX - outerSize / 2,
+                                            newY - outerSize / 2,
+                                            outerSize,
+                                            outerSize
+                                        );
+
+                                        batcher.Draw
+                                        (
+                                            SolidColorTextureCache.GetTexture(outerColor),
```

Comment:
This function will create a lot of textures if the `outerColor` changes.
This might be a good opportunity to add the support for RGBA color drawing function instead of UO color.

Reference and requirements:
Function: https://github.com/ClassicUO/ClassicUO/blob/impl/ecs/src/ClassicUO.Renderer/Batcher2D.cs#L123
Shader: https://github.com/ClassicUO/ClassicUO/blob/impl/ecs/src/ClassicUO.Renderer/shaders/IsometricWorld.fx